src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -125,5 +140,59 @@ private static Summary[] RunWithExceptionHandling(Func<Summary[]> run)
                 return new[] { Summary.NothingToRun(e.Message, string.Empty, string.Empty) };
             }
         }
+
+        private static Summary RunWithDirtyAssemblyResolveHelper(IEnumerable<Type> types, IEnumerable<Assembly> assemblies, IConfig config, string[] args)
+        {
+            var logger = config.GetNonNullCompositeLogger();
+            var userInteraction = new UserInteraction();
+            var (isParsingSuccess, parsedConfig, options) = ConfigParser.Parse(args, logger, config);
+            if (!isParsingSuccess) // invalid console args, the ConfigParser printed the error
+                return null;
+
+            if (options.PrintInformation)
+            {
+                logger.WriteLine(HostEnvironmentInfo.GetInformation());
+                return null;
+            }
+
+            var effectiveConfig = ManualConfig.Union(config, parsedConfig);
+            var (allTypesValid, allAvailableTypesWithRunnableBenchmarks) = TypeFilter.GetTypesWithRunnableBenchmarks(types, assemblies, logger);
+            
+            if (!allTypesValid) // there were some invalid and TypeFilter printed errors
+                return null;
+
+            if (allAvailableTypesWithRunnableBenchmarks.IsEmpty())
+            {
+                userInteraction.PrintNoBenchmarksError(logger);
+                return null;
+            }
+
+            if (options.ListBenchmarkCaseMode != ListBenchmarkCaseMode.Disabled)
+            {
+                BenchmarkCasesPrinter.PrintList(logger, effectiveConfig, allAvailableTypesWithRunnableBenchmarks, options);
+                return null;
+            }
+
+            var BenchmarkRunInfos = new List<BenchmarkRunInfo[]>();
+
+            foreach (Type type in types)
+            {
+                var BenchmarkRunInfo = options.UserProvidedFilters
+                               ? TypeFilter.Filter(effectiveConfig, allAvailableTypesWithRunnableBenchmarks)
```

Comment:
repeating this operation in a loop would end up in duplicates

Reply:
but Types will hold different types. I dont understand how this will create duplicate of benchmarkruninfo.

Reply:
in case `options.UserProvidedFilters` is true, `TypeFilter.Filter(effectiveConfig, allAvailableTypesWithRunnableBenchmarks)` is executed for every `Type` in `types. Which means that if user has provided filter, we filter `allAvailableTypesWithRunnableBenchmarks` types-many times and add the result to a list than many times